### PR TITLE
Feature/create database layer

### PR DESCRIPTION
**Pull request by khlopkov**
In this pull request were moved commits related with creating database layer.
Old will be closed.

### CODE REVIEW

File: Arcadia.Ask/Storage/Questions/QuestionStorage.cs [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+                throw new QuestionNotFoundException(questionId);
+            }
+
+            this.DetachEntity(foundQuestion);
+            return foundQuestion;
+        }
+
+        public async Task<QuestionEntity> UpsertQuestion(QuestionEntity question)
+        {
+            await this.dbCtx.AddAsync(question);
```
Comment (santee) [inline Arcadia.Ask/Storage/Questions/QuestionStorage.cs]: please check if add returns the same entry as db.Entry() call

Comment (khlopkov) [reply Arcadia.Ask/Storage/Questions/QuestionStorage.cs]: [https://docs.microsoft.com/en-us/ef/core/api/microsoft.entityframeworkcore.dbcontext.addasync#Microsoft_EntityFrameworkCore_DbContext_AddAsync_System_Object_System_Threading_CancellationToken_](AddAsync)

> Begins tracking the given entity, and any other reachable entities that are not already being tracked, in the Added state such that they will be inserted into the database when SaveChanges() is called.

As i understood, yes

File: Arcadia.Ask/Storage/Questions/QuestionStorage.cs [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        public async Task<QuestionEntity> DownvoteQuestion(Guid questionId, Guid userId)
+        {
+            return await this.VoteForQuestion(questionId, userId, VoteStatus.Downvoted);
+        }
+
+        private async Task<QuestionEntity> FindQuestionEntityByIdAsync(Guid questionId)
+        {
+            return await this.dbCtx.Questions
+                .Where(q => q.QuestionId == questionId)
+                .Include(q => q.Votes)
```
Comment (santee) [inline Arcadia.Ask/Storage/Questions/QuestionStorage.cs]: why don't use AsNoTracking

Comment (khlopkov) [reply Arcadia.Ask/Storage/Questions/QuestionStorage.cs]: This is private method for getting attached entity. For example when i getting question to update or remove it then.

**Pull request merged**